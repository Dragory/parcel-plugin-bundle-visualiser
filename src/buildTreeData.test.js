const buildTreeData = require('./buildTreeData');

jest.unmock('path');


describe('./src/buildTreeData', ()=>{

  it('should return an empty tree for an empty bundle', ()=>{
    const mockParcelBundle = {
      childBundles: new Set()
    };
    const tree = buildTreeData(mockParcelBundle);
    expect(tree).toMatchSnapshot();
  });

  it('should generate tree data', ()=>{
    const mockParcelBundle = {
      childBundles: new Set([
        {  
          name:'mock bundle 1',
          totalSize:400,
          childBundles: new Set(),
          assets:new Set([
            {
              name:'module/folder1/asset1',
              bundledSize:120
            },
            {
              name:'module/folder1/asset2',
              bundledSize:100
            }
          ])
        },
        {  
          name:'mock bundle 2',
          totalSize:200,
          childBundles: new Set(),
          assets:new Set([
            {
              name:'module/folder1/asset2',
              bundledSize:100
            }
          ])
        }
      ])
    };
    const tree = buildTreeData(mockParcelBundle);
    expect(tree).toMatchSnapshot();
  });

});