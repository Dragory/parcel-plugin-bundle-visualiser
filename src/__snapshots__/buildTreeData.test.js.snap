// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./src/buildTreeData should generate tree data 1`] = `
Object {
  "groups": Array [
    Object {
      "groups": Array [
        Object {
          "groups": Array [
            Object {
              "groups": Array [
                Object {
                  "label": "asset1",
                  "weight": 120,
                },
                Object {
                  "label": "asset2",
                  "weight": 100,
                },
              ],
              "label": "folder1",
              "weight": 220,
            },
          ],
          "label": "module",
          "weight": 220,
        },
      ],
      "label": "mock bundle 1",
      "weight": 400,
    },
    Object {
      "groups": Array [
        Object {
          "groups": Array [
            Object {
              "groups": Array [
                Object {
                  "label": "asset2",
                  "weight": 100,
                },
              ],
              "label": "folder1",
              "weight": 100,
            },
          ],
          "label": "module",
          "weight": 100,
        },
      ],
      "label": "mock bundle 2",
      "weight": 200,
    },
  ],
}
`;

exports[`./src/buildTreeData should return an empty tree for an empty bundle 1`] = `
Object {
  "groups": Array [],
}
`;
